### PR TITLE
Add Ayaneo KUN

### PR DESCRIPTION
**Pull request by 0SkillAllLuck**
Add the Ayaneo KUN. Seems like most stuff is the same from the Ayaneo 2S series.

The full list of keycodes is:
- Back left: 97,125,185
- Back right: 97,125,186
- Top left: 316
- Top Right: 97,125
- Bottom Left: 314 and 315
- Bottom Right: 97,125,187 and 32,125

### CODE REVIEW

**Comment by pastaq** [comment]
Thanks for adding this. A couple notes.
- aya_kun doesn't really match the current naming convention. If prefer if we keep that consistent and just add the next generation. Additional models might come out that use the same hwid and buttons
- it seems like a waste to use the Aya button for mode on the kun since that's already mapped to another physical button.
- I suspect the top right might have an unmapped scancode? It seems strange that it's only 97, 125. Evtest or dmesg should identify if that's correct. If not, you'll need to check for and clear the button that is mapped to top right in each of the three other codes that have 97, 125 or they could incidentally fire two events when pressed.
- for clarification only, bottom left and bottom right are two buttons each and not one button with two modes?

**Comment by 0SkillAllLuck** [comment]
evtest only shows 97 and 125, dmesg shows a `Unknown key released (translated set 2, code 0x69 on isa0060/serio0).` so I guess it is sending a numlock event as well. Any idea how to map this @pastaq?

And yeah there are two buttons on each side on the bottom, left has the X-Box **view** and **menu** buttons. The right side has the big **AYA** button and a small **=** button.

Thought to bind AYA to the MODE button to make it similar to how it works under windows. But wasting a button would be stupid I guess. But using the two top buttons for MODE and QAM works as well.

**Comment by 0SkillAllLuck** [comment]
So thinking the following layout:
- LC: Screenshot
- RC: OSK
- T (top right): QAM
- AYA: Launch Chimera
- =: ESC

**Comment by pastaq** [comment]
Looking at the images here, I just want to make sure we're on the same page with each button. 

The top two buttons are labeled LC/RC
The back buttons are labeled LC1/LC2 & RC1/RC2
the front has the traditional Aya space and what we've been calling "Aya small" 

There should be 8 buttons total, unless they changed the design?

**Comment by 0SkillAllLuck** [comment]
Haven't looked at the back buttons yet, but they worked ingame so guess they are already mapped somehow.

The complete left side of the devices front buttons seem to already use standard keycodes (LC: 316, View: 314, Menu: 315). 

Top right is the "T" button, so mapped it to the QAM, but the keycodes are 97, 125, 185 after mapping them.

**Comment by pastaq** [comment]
> Haven't looked at the back buttons yet, but they worked ingame so guess they are already mapped somehow.

That's interesting. Can you see what they produce under `evtest` ?

> The complete left side of the devices front buttons seem to already use standard keycodes (LC: 316, View: 314, Menu: 315).
By LC do you mean the front face button or the second shoulder button inboard to the LB? I believe LC/R stands for Left/Right Center 

> Top right is the "T" button, so mapped it to the QAM, but the keycodes are 97, 125, 185 after mapping them.

The face button in white?

**Comment by pastaq** [comment]
ayaneo
Lets use this for common reference

**Comment by 0SkillAllLuck** [comment]
So it would be:
LC: Screenshot
RC: OSK
T: QAM
AYAspace: Open Chimera
AYA small: ESC

Or after change from discord:
LC: Screenshot
RC: OSK
T: Open Chimera
AYAspace: MODE (maybe allow remap in config)
AYA small: QAM

File: src/handycon/handhelds/aya_gen8.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    elif active_keys == [] and seed_event.code in [97, 125, 186] and button_on == 0 and button4 in handycon.event_queue:
+        await handycon.handle_key_up(seed_event, button4)
+
+    # BUTTON 5: AYAspace 
+    if active_keys == [97, 125, 187] and button_on == 1 and button5 not in handycon.event_queue:
+        await handycon.handle_key_down(seed_event, button5)
+    elif active_keys == [] and seed_event.code in [97, 125, 187] and button_on == 0 and button5 in handycon.event_queue:
+        await handycon.handle_key_up(seed_event, button5)
+
+    # BUTTON 6: T Button
```
Comment (pastaq) [inline src/handycon/handhelds/aya_gen8.py:64]: This comment conflicts with line 46. One of them should be Aya small.

**Review by pastaq — CHANGES_REQUESTED** [review]

File: src/handycon/handhelds/aya_gen8.py (hunk 1 above)
Comment (0SkillAllLuck) [reply src/handycon/handhelds/aya_gen8.py:64]: Missed that when changing to the new layout. Line 46 should be the small = button

**Comment by pastaq** [comment]
Looks good. I'll work with AYANEO to find a way to map the back buttons. If they aren't willing we might need to find a way to reverse engineer it. I suspect a kernel driver and settings daemon would be needed for full support, which would be out of scope for this project, but it's something that might be necessary considering every device since the AIR has the ability to adjust vibration intensity and some button mapping of some kind.

**Pull request merged**